internal/services/compute/protected_settings_from_key_vault.go

```diff
@@ -37,39 +39,67 @@ func protectedSettingsFromKeyVaultSchema(conflictsWithProtectedSettings bool) *p
 	}
 }
 
-func expandProtectedSettingsFromKeyVault(input []interface{}) *compute.KeyVaultSecretReference {
+func expandProtectedSettingsFromKeyVaultOld(input []interface{}) *compute.KeyVaultSecretReference {
 	if len(input) == 0 {
 		return nil
 	}
 
 	v := input[0].(map[string]interface{})
 
 	return &compute.KeyVaultSecretReference{
-		SecretURL: utils.String(v["secret_url"].(string)),
+		SecretURL: pointer.To(v["secret_url"].(string)),
 		SourceVault: &compute.SubResource{
 			ID: utils.String(v["source_vault_id"].(string)),
 		},
 	}
 }
 
-func flattenProtectedSettingsFromKeyVault(input *compute.KeyVaultSecretReference) []interface{} {
-	if input == nil {
+func expandProtectedSettingsFromKeyVault(input []interface{}) *virtualmachinescalesets.KeyVaultSecretReference {
+	if len(input) == 0 {
 		return nil
 	}
 
-	secretUrl := ""
-	if input.SecretURL != nil {
-		secretUrl = *input.SecretURL
+	v := input[0].(map[string]interface{})
+
+	return &virtualmachinescalesets.KeyVaultSecretReference{
+		SecretUrl: v["secret_url"].(string),
+		SourceVault: virtualmachinescalesets.SubResource{
+			Id: utils.String(v["source_vault_id"].(string)),
+		},
+	}
+}
+
+func flattenProtectedSettingsFromKeyVaultOld(input *compute.KeyVaultSecretReference) []interface{} {
+	if input == nil {
+		return nil
 	}
 
 	sourceVaultId := ""
-	if input.SourceVault != nil && input.SourceVault.ID != nil {
+	if input.SourceVault.ID != nil {
 		sourceVaultId = *input.SourceVault.ID
 	}
 
 	return []interface{}{
 		map[string]interface{}{
-			"secret_url":      secretUrl,
+			"secret_url":      input.SecretURL,
+			"source_vault_id": sourceVaultId,
+		},
+	}
+}
+
+func flattenProtectedSettingsFromKeyVault(input *virtualmachinescalesets.KeyVaultSecretReference) []interface{} {
+	if input == nil {
+		return nil
+	}
+
+	sourceVaultId := ""
+	if input.SourceVault.Id != nil {
```

Comment:
potential crash here too:

```suggestion
	if input.SourceVault != nil && input.SourceVault.Id != nil {
```

Reply:
`SourceVault` isn't a pointer